client/asset/eth/multirpc.go

```diff
@@ -106,6 +107,9 @@ func (p *provider) setTip(header *types.Header) {
 
 // cachedTip retrieves the last known best header.
 func (p *provider) cachedTip() *types.Header {
+	p.tip.RLock()
+	defer p.tip.RUnlock()
```

Comment:
The RLock/RUnlock are still below though.  Mean to move them up here (remove them from below)?

Reply:
Yes this looks like a mistake. Fixing.

Reply:
Thanks for removing the double lock, but... why does it need to lock up here?  Isn't `p.ws` immutable?

Reply:
I think so, moving it down.